src/sql/workbench/services/notebook/browser/notebookViews/notebookViewsExtension.ts

```diff
@@ -83,14 +87,14 @@ export class NotebookViewsExtension extends NotebookExtension<INotebookViewMetad
 					this.setCellMetadata(cell, meta);
 				});
 			}
-
-			this.setNotebookMetadata(this.notebook, this._metadata);
 		}
 
 		if (guid === this._metadata.activeView) {
 			this._metadata.activeView = undefined;
 		}
 
+		this._metadata = Object.assign({}, this._metadata);
```

Comment:
What are you trying to accomplish here by copying to a new object? If metadata contains other objects in it then those won't have new ones created - you'll still be referencing the old ones. Do you mean to do a deepClone here instead?

Reply:
Yes, I was looking to get a new reference for metadata. deepClone fails here, I think metadata's structure is too complex. Not sure if there's a circular dependency or something

Reply:
It seems like you should probably just not do anything here then until you figure it out - having a partially cloned object is potentially worse as you could end up with inconsistent and confusing state depending on what things are being updated.